RecoJets/JetProducers/plugins/ECFAdder.cc

```diff
@@ -2,21 +2,56 @@
 #include "fastjet/PseudoJet.hh"
 #include "fastjet/ClusterSequence.hh"
 #include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
 
 ECFAdder::ECFAdder(const edm::ParameterSet& iConfig) :
   src_(iConfig.getParameter<edm::InputTag>("src")),
   src_token_(consumes<edm::View<reco::Jet>>(src_)),
   Njets_(iConfig.getParameter<std::vector<unsigned> >("Njets")),
+  ecftype_(iConfig.getParameter<std::string>("ecftype")),
   beta_(iConfig.getParameter<double>("beta"))
 {
-  for ( std::vector<unsigned>::const_iterator n = Njets_.begin(); n != Njets_.end(); ++n )
-    {
-      std::ostringstream ecfN_str;
-      ecfN_str << "ecf" << *n;
-      variables_.push_back(ecfN_str.str());
-      produces<edm::ValueMap<float> >(ecfN_str.str().c_str());
-      routine_.push_back(std::auto_ptr<fastjet::contrib::EnergyCorrelator> ( new fastjet::contrib::EnergyCorrelator( *n, beta_, fastjet::contrib::EnergyCorrelator::pt_R ) ));
-    }  
+  if ( iConfig.exists("alpha") ) {
```

Comment:
"alpha" always exist, since you are now using fillDescription()
If you want to treat separately old configs which did not have alpha explicitely set, you must look for a different logic